dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/CompanyOwnershipChecker.kt

```diff
@@ -56,4 +59,44 @@ class CompanyOwnershipChecker(
         val companyId = dataMetaInformationManager.getDataMetaInformationByDataId(dataId).company.companyId
         return isCurrentUserCompanyOwnerForCompany(companyId)
     }
+
+    /**
+     * Method to check whether a given company has at least one company owner
+     * @param companyId the ID of the company
+     * @return a Boolean indicating whether the company has at least one company owner
+     */
+    fun companyExistsAndHasNoOwner(companyId: String): Boolean {
+        return try {
+            val companyOwners = companyRolesControllerApi.getCompanyRoleAssignments(
+                CompanyRole.CompanyOwner, UUID.fromString(companyId),
+            )
+            companyOwners.isEmpty()
+        } catch (clientException: ClientException) {
+            if (clientException.statusCode == HttpStatus.NOT_FOUND.value()) {
+                false
+            } else {
+                throw clientException
+            }
+        }
+    }
+
+    /**
+     * Method to check whether the patch contains only fields that are allowed to be altered by the uploader
```

Comment:
Ich schlage vor zu schreiben "altered by a non-keycloak-admin-user"
anstatt "altered by the uploader".
Denn ein keycloak admin user könnte alle Felder patchen. Für einen keycloak admin user spielt diese Funktion also keine Rolle.

Reply:
done